include/net/buf.h

```diff
@@ -153,6 +153,20 @@ static inline void net_buf_simple_reset(struct net_buf_simple *buf)
 	buf->data = buf->__buf;
 }
 
+
+/**
+ * Clone buffer state, using the same data buffer.
+ *
+ * Initializes a buffer to point to the same data as an existing buffer.
+ * Allows operations on the same data without altering the length and
+ * offset of the original.
+ *
+ * @param clone Clone buffer.
+ * @param original Buffer to clone.
+ */
+void net_buf_simple_clone(struct net_buf_simple *clone,
+			  const struct net_buf_simple *original);
```

Comment:
I'd swap the parameter order since `original` is "the" object you're cloning and `clone` is the output/result of the function. Thoughts? (I'm not completely convinced either way)

Reply:
Either way is fine by me. I used this order to match memcpy, as that's the stdlib equivalent of this, but either is fine by me.

Reply:
Let's swap. The primary object that the APIs operate on is usually the first parameter. Or to put it another way, for an API called `net_buf_simple_<verb>()` the first parameter should be the answer to the question "what do you want to \<verb\>?".